eng/targets/ReadyToRun.targets

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project>
+  
+  <Target Name="_InitializeTargetAssembly">
+    <ItemGroup>
+      <_R2ROptimizeAssembly Include="@(IntermediateAssembly)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_LocateCrossgen2">
+    <PropertyGroup>
+      <_Crossgen2Path>$(NuGetPackageRoot)Microsoft.NETCore.App.crossgen2.win-x64\$(MicrosoftNetCoreAppPackagesVersion)\tools\crossgen2.dll</_Crossgen2Path>
+      <_RunCrossgen2>false</_RunCrossgen2>
+      <_RunCrossgen2 Condition="'$(EnableReadyToRunGeneration)' == 'true' or Exists('$(_Crossgen2Path)')">true</_RunCrossgen2>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="_CalculateCrossgen2Invocations">
+    <ItemGroup>
+      <_R2ROptimizeAssembly Include="@(IntermediateAssembly)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="RunCrossgen2"
+          AfterTargets="CoreCompile"
+          DependsOnTargets="_LocateCrossgen2;_CalculateCrossgen2Invocations"
+          Condition="'$(EnableReadyToRunGeneration)' == 'true'">
+
+    <Message Text='Crossgen2 will be run with argument: $(_Crossgen2Path) %(_R2ROptimizeAssembly.FullPath)'
+             Importance="normal"/>
+  </Target>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NETCore.App.crossgen2.win-x64"  Version="$(MicrosoftNetCoreAppPackagesVersion)" ExcludeAssets="all" PrivateAssets="all" />
```

Comment:
Do we want to include `win-x64` packages unconditionally? What if the build targets ARM?